lib/system/zephyr/mbox.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2025 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * @file	mbox.h
+ * @brief	Zephyr mbox primitives for libmetal.
+ *
+ */
+
+#ifndef __METAL_MBOX_ZEPHYR__H__
+#define __METAL_MBOX_ZEPHYR__H__
+
+#include <metal/assert.h>
+#include <metal/sys.h>
+#include <zephyr/drivers/mbox.h>
```

Comment:
Is there any use case where Zephyr would call this metal API? How do you address the legacy IPM API?
I think we should forbid this API for Zephyr, as a Zephyr application should use the Zephyr mailbox or IPM API instead.

Reply:
I would like to avoid such implementation but having a more portable one (from my POV).
Having a generic API header file that declares API with one .c  file per system seems to me a better way. 
in other term I would like to avoid the use of ` #include <metal/system/@PROJECT_SYSTEM@/mbox.h>`